### PR TITLE
lookup tag before delete a manifest.

### PR DESCRIPTION
**Pull request by wutongjie23hao**
we should first lookup tag before delete a manifest. Thus fix bug manifests unknown and the tag is not deleted by.

### CODE REVIEW

File: registry/handlers/manifests.go [hunk 1]
```diff
@@ -443,7 +443,8 @@ func (imh *manifestHandler) DeleteManifest(w http.ResponseWriter, r *http.Reques
 		return
 	}
 
-	err = manifests.Delete(imh, imh.Digest)
+	tagService := imh.Repository.Tags(imh)
+	referencedTags, err := tagService.Lookup(imh, distribution.Descriptor{Digest: imh.Digest})
```
Comment (zhouhaibing089) [inline registry/handlers/manifests.go]: The below `switch err` is not required for tag lookup api, right?

Comment (wutongjie23hao) [reply registry/handlers/manifests.go]: yes, but if i do without ``swith err`` , than i dont pass the test case.

**Comment by stevvooe** [comment]
What if the tag is deleted in the middle of deleting the manifest?

**Comment by wutongjie23hao** [comment]
when we decide delete a tag, we only should assure that the tag is deleted. If the manifest is not deleted, gc will delete it. But if we delete the manifest at first, the tag is not deleted at all.

**Comment by milosgajdos** [comment]
Closing as wildly outdated. Please feel free to reopen if need be.

**Pull request closed without merging**